packages/gui/src/components/nfts/NFTPreview.tsx

```diff
@@ -162,22 +470,314 @@ export default function NFTPreview(props: NFTPreviewProps) {
+      stopVideo();
+    }
+    if (thumbnail.images) {
+      clearTimeout(loopImageInterval);
     }
   }
 
+  function isDocument() {
+    return (
+      [
+        'pdf',
+        'docx',
+        'doc',
+        'xls',
+        'xlsx',
+        'ppt',
+        'pptx',
+        'txt',
+        'rtf',
+      ].indexOf(extension) > -1
+    );
+  }
+
+  function renderCompactIcon() {
+    return (
+      <CompactIconFrame>
+        <CompactIcon />
+        {mimeType().match(/^video/) && <CompactVideoIcon />}
+        {mimeType().match(/^audio/) && <CompactAudioIcon />}
+        {mimeType().match(/^model/) && <CompactModelIcon />}
+        {isDocument() && <CompactDocumentIcon />}
+        {isUnknownType() && <CompactUnknownIcon />}
+        {extension && <CompactExtension>.{extension}</CompactExtension>}
+      </CompactIconFrame>
+    );
+  }
+
+  function renderNftIcon() {
+    if (isDocument()) {
+      return (
+        <>
+          <BlobBg>
+            <DocumentBlobIcon />
+            <img src={isDarkMode ? DocumentPngDarkIcon : DocumentPngIcon} />
+          </BlobBg>
+        </>
+      );
+    } else if (mimeType().match(/^model/)) {
+      return (
+        <>
+          <BlobBg>
+            <ModelBlobIcon />
+            <img src={isDarkMode ? ModelPngDarkIcon : ModelPngIcon} />
+          </BlobBg>
+        </>
+      );
+    } else if (mimeType().match(/^video/)) {
+      return (
+        <>
+          <BlobBg>
+            <VideoBlobIcon />
+            <img src={isDarkMode ? VideoPngDarkIcon : VideoPngIcon} />
+          </BlobBg>
+        </>
+      );
+    } else {
+      return (
+        <>
+          <BlobBg>
+            <UnknownBlobIcon />
+            <img src={isDarkMode ? UnknownPngDarkIcon : UnknownPngIcon} />;
+          </BlobBg>
+        </>
+      );
+    }
+  }
+
+  function isUnknownType() {
+    return (
+      !isDocument() &&
+      !mimeType().match(/^audio/) &&
+      !mimeType().match(/^video/) &&
+      !mimeType().match(/^image/) &&
+      !mimeType().match(/^model/)
+    );
+  }
+
+  function renderElementPreview() {
+    if (isCompact && !mimeType().match(/^image/)) {
+      return renderCompactIcon();
+    }
+
+    const isOfferNft =
+      isOffer &&
+      !mimeType().match(/^video/) &&
+      !mimeType().match(/^audio/) &&
+      !mimeType().match(/^image/);
+
+    const notPreviewNft =
+      !isOffer &&
+      !isPreview &&
+      (mimeType().match(/^model/) || isDocument() || isUnknownType());
+
+    const isPreviewNft =
+      !mimeType().match(/^image/) &&
+      !thumbnail.video &&
+      !thumbnail.image &&
+      !mimeType().match(/^audio/) &&
+      isPreview &&
+      !isOffer;
+
+    if (isOfferNft || isPreviewNft || notPreviewNft) {
+      return (
+        <>
+          {renderNftIcon()}
+          {extension && (
+            <ModelExtension isDarkMode={isDarkMode}>
+              .{extension}
+            </ModelExtension>
+          )}
+        </>
+      );
+    }
```

Comment:
For my image NFTs that don't have a file extension in their URLs, this block of code is preventing the images from being displayed.

I'm ok with commenting this out for now and revisiting in a separate PR, but we need those existing image NFTs to display properly.

Reply:
I put showing NFT icon back. (where you uncommented it)
From now we're trying to show the image NFT in <Iframe> if resolved file extension from url is empty.